fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -488,7 +478,14 @@ private CompletableFuture<byte[]> readSchemaAsync(List<Long> bitSetWords) {
 
     public byte[] readSchema(List<Long> bitSetWords) throws IOException {
         BitSet bitSet = BitSet.valueOf(ArrayUtils.toPrimitive(bitSetWords.toArray(new Long[0])));
-        return fieldInfosDataCache.asMap().computeIfAbsent(bitSet, ignore -> context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_GET_SCHEMA, readSchemaAsync(bitSetWords)));
+        // In order to avoid a deadlock, and since the readSchema is idempotent, perform a non-blocking, non-atomic cache
+        // population. There may be a few threads that make calls to FDB, but they should all be returning the same result
+        byte[] value = fieldInfosDataMap.get(bitSet);
+        if (value == null) {
```

Comment:
You have this backwards
```suggestion
        if (value != null) {
```

Reply:
This is the keeper for the first check - the one below has != as it should